### ISSUE TITLE
Server won't boot to any fs but 44.1k on OSX built-in hardware

### ISSUE THREAD

**Issue by mtmccrea** [supercollider/supercollider#1961]
I'm on a fresh install of 3.7.1, though this was a problem on 3.7.0, and 3.7.0-beta1 as well.
On OS X (10.9.5), I have the system sound hardware is set to Built-In Input/Output, at sample rate 44100. I run ```s.options.sampleRate_(48000)```. I boot, then this message posts (note, not as an Error or even a Warning):

> Requested sample rate 48000.000000 was not available - attempting to use sample rate of 44100.000000
> "Built-in Input" Input Device
>    Streams: 1
>       0  channels 2
> 
> "Built-in Output" Output Device
>    Streams: 1
>       0  channels 2
> 
> SC_AudioDriver: sample rate = 44100.000000, driver's block size = 512
> SuperCollider 3 server ready.
> Receiving notification messages from server localhost
> Shared memory server interface initialized

After manually setting the hardware to 48000 in Audio MIDI Setup, SC still thinks 48000 is unavailable and boots to 44100, _changing the hardware back to 44100_. Yikes!

None of the sample rates above 44100 work (and server boots to 44100) and any sample rate under 44100 (8k, 16k, 32k) fails:

> set kAudioDevicePropertyNominalSampleRate error tad!
> could not initialize audio.
> RESULT = 1
> ERROR: server failed to start
> For advice: [http://supercollider.sf.net/wiki/index.php/ERROR:_server_failed_to_start]
> /quit sent

The SR failure aside, I'd argue that booting to another sample rate than requested is bad behavior. I could see why in some cases you'd want to run audio even at a mismatch from the requested SR, but perhaps this should be a flag set when booting the server, or at least post a Warning/Error to get the benefit of error formatting in the post window to catch your attention (the message that's posted could easily be missed in an automated startup routine).

This is a problem on 3.7.1/0/0-beta1. Interestingly, 3.6.6 can set the sample rate of the hardware, though after booting, the post window displays the _previous_ sample rate, as does ```server.sampleRate```, despite the server's _actual_ sample rate, ```s.actualSampleRate```, displaying the requested sample rate.

This is behavior I reported in #768.
This new issue could be related or have been introduced by the fix to #1612, which pertains to assuming a sample rate was set correctly.

Thanks for having a look!

**Comment by miczac** [comment]
I think I ran into the same problem and that's about to get fixed.
Last word is not yet spoken on that but I'll forward the relevant  email from @scztt  on the dev list.

The first build on row might fix it, the second one (in the quoted mail) outputs a list of available devices.
I'd be curious how that looks like on your Mac (what model, BTW?).

```
See if this fixes?
http://supercollider.s3.amazonaws.com/builds/supercollider/supercollider/osx/SC-0ff7a23898591eeae2cf2d526378b1a09a6d76b7.zip 

On Sun, Apr 10, 2016 at 11:19 AM Scott Carver <scott@artificia.org> wrote:
There will be a build here as soon as travis is finished:
http://supercollider.s3.amazonaws.com/builds/supercollider/supercollider/osx/SC-86c8490bcc75eb22c98e8c44b3aad3893758032a.zip 

Test with that in the cases where you're seeing issues? It should print out the supported sample rates as returned by your audio devices - plz send those along, it should help pinpoint things.
 - S
```

**Comment by miczac** [comment]
[PLS note the updates in the msg above]
FWIW, on my mid 2012 MBP I can't set 96k (which I need).
So, I found a workaround: 
just set, for instance
```s.options.outDevice = "Built-in Output";
s.options.inDevice = "Built-in Input";```
but don't  set ```s.options.sampleRate``` at all.
Instead use *Audio Midi Setup* to choose the rate on the device of your choice.
Not convenient, but that works on my machine.
HTH, m.

**Comment by mtmccrea** [comment]
Ah... I didn't see the recent post on the dev list, looks like Scott fixed it.

So the first linked build, which posts the devices and available rates
http://supercollider.s3.amazonaws.com/builds/supercollider/supercollider/osx/SC-0ff7a23898591eeae2cf2d526378b1a09a6d76b7.zip
appears to work at all the sample rates, and on rebooting the server after the ServerOptions sr is set. 
Great!

Also, your workaround works for me on 3.7.1. on my mid-2010 MBP.

Will this be patched into 3.7.1 or 3.7.2?

Thanks to you and @scztt !
